ethcore/sync/src/block_sync.rs

```diff
@@ -523,15 +533,24 @@ impl BlockDownloader {
 		trace!(target: "sync", "Imported {} of {}", imported.len(), count);
 		self.imported_this_round = Some(self.imported_this_round.unwrap_or(0) + imported.len());
 
-		if bad {
-			return Err(BlockDownloaderImportError::Invalid);
-		}
-
 		if self.blocks.is_empty() {
```

Comment:
Is that correct? Seems that if we fail on the last block we are not going to return an error?